components/TemplateCard/index.js

```diff
@@ -58,6 +67,24 @@ export default function TemplateCard({ templateData }) {
     }, 1500);
   };
 
+  const handleDeleteClick = (id) => {
+    deleteTemplate(id);
+    setIsDelete(false)
+  }
+
+  useEffect(() => {
+    const handleWindowClick = (event) => {
+      if (isDelete && deleteRef.current && !deleteRef.current.contains(event.target)) {
+        setIsDelete(false);
+      }
+    }
+
+    document.addEventListener("mousedown", handleWindowClick)
+
+    return () => document.removeEventListener("mousedown", handleWindowClick)
+  }, [isDelete])
```

Comment:
I am note really sure if you really need this useEffect. Have you tried to comment it out and check if everything does what it does (because I tried and saw no difference 😅)? 🤔

Reply:
Ok, that totally makes sense! I think that there might be a solution without a useEffect for this. I just thought that you could also work with an `onClick` handler inside the JSX, but then you need to lift up the state that would make your solution not less complex. So I think it is ok, even though we normally do not work with `addEventListener` in React. [If you want to read further for future improvement, there is a chapter about responding to events in the React docs.](https://react.dev/learn/responding-to-events)